armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -0,0 +1,492 @@
+# Copyright 2019 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""enable component-wise axial expansion for assemblies and/or a reactor"""
+
+from statistics import mean
+from numpy import array
+from armi import runLog
+from armi.reactor.flags import Flags
+
+
+class AxialExpansionChanger:
+    """
+    Axially expand or contract assemblies or an entire core.
+
+    Useful for fuel performance, thermal expansion, reactivity coefficients, etc.
+    """
+
+    def __init__(self, converterSettings: dict):
```

Comment:
Great to see type hints!

Reply:
Didn't see them anywhere else in the framework so I wasn't sure if they were wanted or not. I'll keep this in mind!

Reply:
We like type hints!

They _are_ in ARMI. But ARMI is older than type hints in Python,  so still about 80-90% of the code doesn't have any hints.